staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_cbor.go

```diff
@@ -0,0 +1,35 @@
+package v1
+
+import (
+	"github.com/fxamacker/cbor/v2"
+)
+
+var _ cbor.Marshaler = JSONSchemaPropsOrBool{}
+var _ cbor.Unmarshaler = &JSONSchemaPropsOrBool{}
+
+var (
+	cborFalse = []byte{0xf4}
+	cborTrue  = []byte{0xf5}
+)
+
+func (s JSONSchemaPropsOrBool) MarshalCBOR() ([]byte, error) {
+	if s.Schema != nil {
+		return cbor.Marshal(s.Schema)
+	}
+
+	if s.Schema == nil && !s.Allows {
+		return cborFalse, nil
+	}
+	return cborTrue, nil
+}
+
+func (s *JSONSchemaPropsOrBool) UnmarshalCBOR(data []byte) error {
+	if err := cbor.Unmarshal(data, &s.Allows); err == nil {
+		return nil
+	}
+	if err := cbor.Unmarshal(data, &s.Schema); err != nil {
```

Comment:
Ideally we could distinguish the error condition "input is not a map so it can't be decoded into a `*JSONSchemaProps`" from other conditions, but I don't think that's practical without contributing support for that to the library. It could be worth a comment here.

Reply:
Something to check: does this handle an explicit `null` the same way the JSON implementation does?